theseus/geometry/utils/utils.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+
+class LieGroupTensor(torch.Tensor):
```

Comment:
I am wondering if we need this class to be in a separate utils folder (i.e. do we anticipate more stuff) or we can just have all this in theseus/geometry.

Reply:
I have no preference and I'm OK for this.

Reply:
I think we can keep these in the same folder, as @mhmukadam suggests. On the other hand, we want to discourage users from using these classes directly, so we should rename as `_LieGroupTensor` and `_LieGroupContext`.

Reply:
Agreed. Will make this change.

Reply:
See #93 and I will create a new PR for it.